src/components/routes/Trips/Trips.tsx

```diff
@@ -173,9 +173,9 @@ export default compose(
   graphql(GUEST_CANCEL_BOOKING, {
     props: ({ mutate }: any) => ({
       cancelBooking: async (booking: Booking) => {
-        const { id, currency, checkInDate } = booking;
+        const { id, currency, checkInDate, status } = booking;
         const days = differenceInDays(checkInDate, Date.now());
-        if (currency === Currency.BEE && days >= 7) {
+        if (currency === Currency.BEE && days >= 7 && status === 'guest_paid') {
```

Comment:
would this restrict the `cancel` to only 'guest_paid'? I'm not too familiar with the metamask flow but would we want to allow `cancel` for other statuses further in the process like 'host_approved' as well?

Reply:
One thing to note is that this allows cancelling the booking regardless of status (or, it obeys whatever status constraints have been specified elsewhere); the call being conditioned here is the call to the contract to reclaim funds.

Looking over the statuses defined in the Booking model:

                'started',
                'expired_before_guest_confirmed',
                'guest_confirmed',
                'guest_cancel_initiated',
                'guest_cancelled',
                'guest_rejected_payment',
                'host_rejected',
                'host_approved',
                'host_cancelled',
                'expired_before_host_approved',
                'init_pay_submitted',
                'payment_failed',
                'guest_paid',
                'host_paid',
                'dispute_initiated',
                'disputed',
                'refund_initiated',
                'refunded',
                'completed'

There is a case where status is `host_approved` but the backend hasn't yet picked up on the event to advance the status to `guest_paid`. This is an ambiguous situation; we don't know if the host's `invoice` call actually went through (if we did, status would be `guest_paid`), so we don't know if the Payments contract has any funds for the user to reclaim.

I think that erring on the side of caution here means looking at just `guest_paid`, insofar as (In the event that a user does find a way to hit that edge case) we can handle the refund with an off-line process.

Reply:
(More broadly, all of this is why I want to decouple booking and payment status 😁)

Reply:
ahh i see. so it practically only needs to 'refund' the contract when `guest_paid` since the money actually left the guest wallet at that point. sounds good!

+1 on decoupling the statuses!